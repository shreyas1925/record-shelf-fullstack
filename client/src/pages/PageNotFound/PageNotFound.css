
.container--page{
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    padding: 10px;
    padding-top: 20px;
}

.container--content img{
    height: 400px;
    width: 500px;
}

.container--description {
    text-align: center;
}

.container--description h1 {
  font-size: 40px;
}

.container--description h3 {
  font-size: 30px;
}

.container--description button{
  color: #fff !important;
  padding: 10px 20px;
  background: #39ac31;
  margin: 20px 0;
  display: inline-block;
  border: none;
  cursor: pointer;
}

a{
    color: inherit;
    text-decoration: inherit;
}

h1,h5{
    padding: 5px;
}

.container--description button:hover{
    color: #39ac31 !important;
    padding: 10px 20px;
    background: #fff;
    margin: 20px 0;
    display: inline-block;
    border: 1px solid #39ac31;
    cursor: pointer;

  }
  
