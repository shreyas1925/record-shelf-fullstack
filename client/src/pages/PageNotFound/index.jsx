import React from "react";
import { Link } from "react-router-dom";
import "./PageNotFound.css";

const PageNotFound = () => {
  return (
    <div className="container--page" data-testid='notfound'>
      <div className="container--content">
        <img
          src="https://cdn.dribbble.com/users/285475/screenshots/2083086/dribbble_1.gif"
          alt=""
        />
        <div className="container--description">
          <h1>404</h1>
          <h5>Look like you're lost , return to home </h5>

          <button className="link_404">
            <Link to="/">Go to Home</Link>
          </button>
        </div>
      </div>
    </div>
  );
};

export default PageNotFound;
