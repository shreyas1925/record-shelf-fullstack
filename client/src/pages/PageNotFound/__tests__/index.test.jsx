import { render, screen } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import PageNotFound from '..';

describe('PageNotFound Page Snapshot', () => {
    it('should render forming a snapshot', () => {
        const { asFragment } = render(<BrowserRouter><PageNotFound/></BrowserRouter>);
        expect(asFragment()).toMatchSnapshot(); 
    });
});

describe('PageNotFound Page', () => {
    it('should render page without crashing', () => {
      
        render(<BrowserRouter><PageNotFound/></BrowserRouter>);
        const errorPageElement = screen.getByTestId('notfound');
        expect(errorPageElement).toBeInTheDocument();
    });

    it('should display text after rendering', () => {
        render(<BrowserRouter><PageNotFound/></BrowserRouter>);
        const errorPageElement = screen.getByText('Look like you are lost , return to home');
        expect(errorPageElement).toBeInTheDocument();
    });
});