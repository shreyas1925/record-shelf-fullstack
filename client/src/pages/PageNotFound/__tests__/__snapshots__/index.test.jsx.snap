// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageNotFound Page Snapshot should render forming a snapshot 1`] = `
<DocumentFragment>
  <div
    class="container--page"
    data-testid="notfound"
  >
    <div
      class="container--content"
    >
      <img
        alt=""
        src="https://cdn.dribbble.com/users/285475/screenshots/2083086/dribbble_1.gif"
      />
      <div
        class="container--description"
      >
        <h1>
          404
        </h1>
        <h5>
          Look like you are lost , return to home 
        </h5>
        <button
          class="link_404"
        >
          <a
            href="/"
          >
            Go to Home
          </a>
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
